week2/project/ecommerce/src/components/Products.js

```
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+import ProductCard from './ProductCard.js';
+//import productsData from '../fake-data/all-products.js';
+import '../styles/products.css'
+
+
+function Products({ selectedCategory }) {
+    const [products, setProducts] = useState([]);
+    const [loadingProducts, setLoadingProducts] = useState(false);
+    const [error, setError] = useState(null);
+    /**
+     * https://fakestoreapi.com/products or 
+     * https://fakestoreapi.com/products/category/:selectedCategory 
+     */
+    useEffect(() => {
+        const fetchData = async () => {
+
+            setLoadingProducts(true);
+
+            try {
+                const URL_2 = selectedCategory ? `https://fakestoreapi.com/products/category/${selectedCategory}` : `https://fakestoreapi.com/products`;
+
+                const response = await fetch(URL_2);
+
+                if (!response.ok) {
+                    throw new Error(`Error! status: ${response.status}`);
```

- Instead of throwing error, you can provide appropriate feedback to users when there's a problem with fetching data from the API.

---

Done. I thought it is better to throw an error!